sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Catalog.scala

```diff
@@ -104,7 +104,7 @@ class SimpleCatalog(val conf: CatalystConf) extends Catalog {
     val tableName = getTableName(tableIdent)
     val table = tables.get(tableName)
     if (table == null) {
-      throw new NoSuchTableException
+      throw new AnalysisException("Table not found: " + tableName)
```

Comment:
Does this mean that `NoSuchTableException` is not used anywhere?  We should probably get rid of it then.  Though, I think I'd have a slight preference for keeping a special exception internally.  The point of this extra control flow is so that we can attach line position information to this specific exception in the analyzer.